hermes/probe/create/create_file_test.go

```diff
@@ -0,0 +1,140 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	metricpb "github.com/google/cloudprober/metrics/proto"
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	"github.com/googleinterns/step224-2020/hermes/probe/mock"
+	journalpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+func TestFileName(t *testing.T) {
+	tests := []struct {
+		file     RandomHermesFile
+		wantName string
+		wantErr  string
+	}{
+		{RandomHermesFile{51, 12}, "", "The file ID provided 51 wasn't in the required range [1,50]"},
+		{RandomHermesFile{0, 50}, "", "The file ID provided 0 wasn't in the required range [1,50]"},
+		{RandomHermesFile{3, 100}, "Hermes_03", ""},
+		{RandomHermesFile{12, 100}, "Hermes_12", ""},
+		{RandomHermesFile{3, 0}, "", "The file size provided 0 is not a positive number as required"},
+		{RandomHermesFile{3, 1001}, "", "The file size provided 1001 bytes exceeded the limit 1000 bytes"},
+	}
+
+	for _, test := range tests {
+		got, err := test.file.FileName()
+		if err == nil {
+			if test.wantErr != "" {
+				t.Errorf("{%v, %v}.FileName() gave wrong error expected %s, got nil", test.file.ID, test.file.Size, test.wantErr)
+			}
+			if test.wantName != got[0:9] {
+				t.Errorf("{%v, %v}.FileName() failed expected prefix %s, got %s", test.file.ID, test.file.Size, test.wantName, got[0:9])
+			}
+		} else {
+			if got != "" {
+				t.Errorf("{%v, %v}.FileName() failed expected empty string, got %s", test.file.ID, test.file.Size, got)
+			}
+			if err.Error() != test.wantErr {
+				t.Errorf("{%v, %v}.FileName() gave wrong error expected %s, got %s", test.file.ID, test.file.Size, test.wantErr, err.Error())
+			}
+		}
+	}
+}
+
+func TestChecksum(t *testing.T) {
+	f := RandomHermesFile{11, 100}
```

Comment:
I would structure this test as a table-driven test:

testCases := []struct{
   f *RandomHermesFile,
   want string
}{
  {&RandomHermesFile{11, 100}, "0xaa ...."},
  {&RandomHermesFile{13, 1000}, "0xaa ...."},
...
}

Reply:
@deniadelcid raised the same issue https://github.com/googleinterns/step224-2020/pull/57#discussion_r489391429 and we discussed it in our POD meeting. Changed the test as per what was suggested -  it's not table driven but should hopefully be more readable than previously